convert2rhel/subscription.py

```diff
@@ -164,25 +164,25 @@ def hide_password(cmd):
 
 
 def replace_subscription_manager():
-    """Remove the original and install the RHEL subscription-manager packages."""
+    """Remove any previously installed subscription-manager packages and install the RHEL ones.
+    
+    Make sure the system is unregistered before removing the subscription-manager as not doing so would leave the
+    system to be still registered on the server side, making it dificult for an admin to unregister it afterwards.
+    """
     if not os.path.isdir(SUBMGR_RPMS_DIR) or not os.listdir(SUBMGR_RPMS_DIR):
-        loggerinst.critical("The %s directory does not exist or is empty."
-                            " Using the subscription-manager is not documented"
-                            " yet. Please use the --disable-submgr option."
-                            " Read more about the tool usage in the article"
-                            " https://access.redhat.com/articles/2360841."
-                            % SUBMGR_RPMS_DIR)
-        return
+        loggerinst.critical("The %s directory does not exist or is empty." % SUBMGR_RPMS_DIR)
 
+    unregister_system()
     remove_original_subscription_manager()
     install_rhel_subscription_manager()
 
 
 def remove_original_subscription_manager():
-    loggerinst.info("Removing non-RHEL subscription-manager packages.")
+    loggerinst.info("Removing subscription-manager packages.")
     # python3-subscription-manager-rhsm, dnf-plugin-subscription-manager, subscription-manager-rhsm-certificates, etc.
-    submgr_pkgs = pkghandler.get_installed_pkgs_w_different_fingerprint(
-        system_info.fingerprints_rhel, "*subscription-manager*")
+    submgr_pkgs = pkghandler.get_installed_pkg_objects("*subscription-manager*")
+    # Satellite-server related package
+    submgr_pkgs += pkghandler.get_installed_pkg_objects("katello*")
```

Comment:
We notify the user that we are removing Satellite related packages? 

Also need to update L189 with telling the user weäre removing sub mgr packages, or have a separate confirmation about the `katello` packages

Reply:
+1